src/Microsoft.Health.CosmosDb/Features/Health/CosmosHealthCheck.cs

```diff
@@ -0,0 +1,80 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.Documents;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.CosmosDb.Configs;
+using Microsoft.Health.CosmosDb.Features.Storage;
+using Microsoft.Health.Extensions.DependencyInjection;
+
+namespace Microsoft.Health.CosmosDb.Features.Health
+{
+    public abstract class CosmosHealthCheck : IHealthCheck
+    {
+        private readonly IScoped<IDocumentClient> _documentClient;
+        private readonly CosmosDataStoreConfiguration _configuration;
+        private readonly CosmosCollectionConfiguration _cosmosCollectionConfiguration;
+        private readonly IDocumentClientTestProvider _testProvider;
+        private readonly ILogger<CosmosHealthCheck> _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CosmosHealthCheck"/> class.
+        /// </summary>
+        /// <param name="documentClient">The document client factory/</param>
+        /// <param name="configuration">The CosmosDB configuration.</param>
+        /// <param name="namedCosmosCollectionConfigurationAccessor">The IOptions accessor to get a named version.</param>
+        /// <param name="collectionConfigurationName"> Name to get corresponding collection configuration</param>
+        /// <param name="testProvider">The test provider</param>
+        /// <param name="logger">The logger.</param>
+        public CosmosHealthCheck(
+            IScoped<IDocumentClient> documentClient,
+            CosmosDataStoreConfiguration configuration,
+            IOptionsSnapshot<CosmosCollectionConfiguration> namedCosmosCollectionConfigurationAccessor,
+            string collectionConfigurationName,
+            IDocumentClientTestProvider testProvider,
+            ILogger<CosmosHealthCheck> logger)
+        {
+            EnsureArg.IsNotNull(documentClient, nameof(documentClient));
+            EnsureArg.IsNotNull(configuration, nameof(configuration));
+            EnsureArg.IsNotNull(namedCosmosCollectionConfigurationAccessor, nameof(namedCosmosCollectionConfigurationAccessor));
+            EnsureArg.IsNotNullOrWhiteSpace(collectionConfigurationName, nameof(collectionConfigurationName));
+            EnsureArg.IsNotNull(testProvider, nameof(testProvider));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _documentClient = documentClient;
+
+            EnsureArg.IsNotNull(_documentClient, optsFn: options => options.WithMessage("Factory returned null."));
```

Comment:
Isn't this already checked by https://github.com/Microsoft/fhir-server/pull/361/files#diff-8b7f39e471a45c4b5fe6f5db9d6b4662R45?

Reply:
I copy pasted this from the previous health check ctor. Will remove it.